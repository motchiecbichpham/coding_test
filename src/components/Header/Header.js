import React from 'react';
import {View, Text,StyleSheet} from 'react-native';
import {getStatusBarHeight} from 'react-native-status-bar-height';
import {COLORS} from '../../utils/colors';
const HEADER_HEIGHT = 45 + 8 + getStatusBarHeight(false);
const Header = ({ title }) => {
  return (
    <View style={styles.header}>
      <View style={styles.titleWrapper}>
        <Text
          numberOfLines={1}
          ellipsizeMode="tail"
          style={styles.textTitle}>
          {title}
        </Text>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  header: {
    paddingHorizontal: 16,
    flexDirection: 'row',
    backgroundColor: COLORS.white,
    paddingTop: getStatusBarHeight(false),
    borderBottomColor: COLORS.brightGray,
    borderBottomWidth: 1,
    alignItems: 'center',
    paddingVertical: 8,
    height: HEADER_HEIGHT,
  },
  titleWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  textTitle: {
    flex: 1,
    fontWeight: 'bold',
    fontSize: 18,
    textAlign: 'center',
    color: COLORS.darkCharcoal,
  },
})
export default Header;
