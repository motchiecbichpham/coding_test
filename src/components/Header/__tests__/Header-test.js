import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

const headerProps = {
  title: 'header',
};
test('render header', () => {
  const tree = renderer.create(<Header {...headerProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
