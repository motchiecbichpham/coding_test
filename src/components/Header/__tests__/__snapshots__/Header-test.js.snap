// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#ECF0F1",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 73,
      "paddingHorizontal": 16,
      "paddingTop": 20,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#333333",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      header
    </Text>
  </View>
</View>
`;
